File: solutions_21-30/Euler_30.py
## Sum of numbers that can be written as the sum of their fifth power. ##

import timeit


start = timeit.default_timer()

def euler_30():
    trial = [str(i) for i in range(2,1000000)]
    total = 0
    answer = []
    for i in trial:
        for v in i:
            total += int(v)**5
        if total == int(i):
            answer.append(int(i))
        total = 0
    return sum(answer)


print "Answer: %s" % euler_30()
stop = timeit.default_timer()
print "Time: %s" % str(stop - start)

File: Perl/perl_solutions_21-30/Euler_30.pl
use strict;
use warnings;
use Time::HiRes qw( clock );


sub sum_fifthpow
{
    my $total = 0;
    for my $i (2..1000000) {
        my $sum = 0;
        for my $j (split(//, $i)) {
            $sum += $j**5;
        }
        if ($sum eq $i) {
            $total += $i;
        }
    }
    return $total;
}


my $start = clock();
my $answer = sum_fifthpow();
my $stop = clock();

print "Answer: $answer\n";
printf "Time: %.5f\n", $stop - $start;
